packages/shared/src/SpotForcedWithdrawalRequest.ts

```diff
@@ -0,0 +1,36 @@
+import { Interface } from '@ethersproject/abi'
+import { StarkKey } from '@explorer/types'
+
+const coder = new Interface([
+  'function fullWithdrawalRequest(uint256 ownerKey, uint256 vaultId)',
+])
+
+export interface SpotForcedWithdrawalRequest {
+  ownerKey: StarkKey
+  vaultId: bigint
+}
+
+export function decodeSpotForcedWithdrawalRequest(
+  data: string
+): SpotForcedWithdrawalRequest | undefined {
+  try {
+    const decoded = coder.decodeFunctionData('fullWithdrawalRequest', data)
+    /* eslint-disable @typescript-eslint/no-unsafe-argument */
+    return {
+      ownerKey: StarkKey.from(decoded.starkKey),
+      vaultId: BigInt(decoded.positionId),
+    }
+    /* eslint-enable @typescript-eslint/no-unsafe-argument */
+  } catch {
+    return
```

Comment:
Wouldn't we want to log the error here?

Reply:
I do not think it will give any value, returning undefined is just fine